File: frontend/src/types/session.ts
export type InteractionMode = 'simulator' | 'manual' | 'replay';
export type DifficultyLevel = 'easy' | 'medium' | 'hard';
export type SessionStatus = 'active' | 'completed';

export interface TranscriptMessage {
  role: 'customer' | 'agent';
  content: string;
}

export interface SessionConfig {
  interaction_mode: InteractionMode;
  industry: string;
  product: string;
  issue_type: string;
  difficulty: DifficultyLevel;
  customer_persona: string;
  customer_mood: string;
  preloaded_transcript?: TranscriptMessage[] | null;
}

export interface Session extends SessionConfig {
  id: string;
  agent_id: string;
  status: SessionStatus;
  created_at: string;
  post_interaction_summary?: any;
  replay_timeline?: any[];
}

File: backend/schemas/session.py
from datetime import datetime
from enum import Enum
from pydantic import BaseModel, Field

class InteractionMode(str, Enum):
    SIMULATOR = "simulator"
    MANUAL = "manual"
    REPLAY = "replay"

class DifficultyLevel(str, Enum):
    EASY = "easy"
    MEDIUM = "medium"
    HARD = "hard"

class SessionStatus(str, Enum):
    ACTIVE = "active"
    COMPLETED = "completed"

class TranscriptMessage(BaseModel):
    role: str
    content: str

class SessionCreate(BaseModel):
    interaction_mode: InteractionMode
    industry: str = Field(..., min_length=1)
    product: str = Field(..., min_length=1)
    issue_type: str = Field(..., min_length=1)
    difficulty: DifficultyLevel = DifficultyLevel.MEDIUM
    customer_persona: str = Field(..., min_length=1)
    customer_mood: str = Field(..., min_length=1)
    preloaded_transcript: list[TranscriptMessage] | None = None

class SessionResponse(SessionCreate):
    id: str
    agent_id: str
    status: SessionStatus
    created_at: datetime
    post_interaction_summary: dict | None = None
    replay_timeline: list[dict] | None = None

    class Config:
        from_attributes = True

class SessionUpdate(BaseModel):
    status: SessionStatus | None = None
